crates/solidity/inputs/schema/grammar/topics/05-expressions/01-base-expressions.yml

```diff
@@ -1,123 +1,276 @@
 # yaml-language-server: $schema=../../../../../../codegen/schema/generated/topic.schema.json
 
 - name: "Expression"
-  kind: "Rule"
+  kind: "PrecedenceParser"
   versioned:
-    # These are versioned so that we generate different code for the exponentation operator.
-    # This is temporary until we change the schema for pratt parsers
     0.4.11:
-      parserType: "Precedence"
-      choice:
-        - reference: "AssignmentExpression"
-        - reference: "ConditionalExpression"
-        - reference: "OrExpression"
-        - reference: "AndExpression"
-        - reference: "EqualityComparisonExpression"
-        - reference: "OrderComparisonExpression"
-        - reference: "BitOrExpression"
-        - reference: "BitXOrExpression"
-        - reference: "BitAndExpression"
-        - reference: "ShiftExpression"
-        - reference: "AddSubExpression"
-        - reference: "MulDivModExpression"
-        - reference: "ExponentiationExpression"
-        - reference: "UnarySuffixExpression"
-        - reference: "UnaryPrefixExpression"
-        - reference: "FunctionCallExpression"
-        - reference: "MemberAccessExpression"
-        - reference: "IndexAccessExpression"
+      operators:
+        - name: "AssignmentExpression"
+          model: "BinaryLeftAssociative"
+          choice:
+            - terminal: "="
+            - terminal: "|="
+            - terminal: "^="
+            - terminal: "&="
+            - terminal: "<<="
+            - terminal: ">>="
+            - terminal: ">>>="
+            - terminal: "+="
+            - terminal: "-="
+            - terminal: "*="
+            - terminal: "/="
+            - terminal: "%="
+
+        - name: "ConditionalExpression"
+          model: "UnarySuffix"
+          sequence: # This subsequence forms the correct CST for the suffix expression parser
+            - terminal: "?"
+            - reference: "Expression"
+            - terminal: ":"
+            - reference: "Expression"
+
+        - name: "OrExpression"
```

Comment:
In this new model, I'm planning on using the same name for similar operators. For example, `OrExpression`, `AndExpression`, `EqualityComparisonExpression`, etc... can all be  `BinaryExpression`.

- The grammar would still handle associativity correctly.
- Always the same CST structure. This makes us produce fewer kinds, and simpler trees to navigate. They have no other syntactic difference.
- In Semantic analysis, we would make that distinction and surface the operator semantics based on the operator node kind directly.

Thoughts?

Reply:
It would still work.